src/modules/pm.c

```diff
@@ -142,28 +135,52 @@ static void publish_susp_req(const bool new) {
     M_PUB(suspend_req);
 }
 
+static void acquire_lock(void) {
+	sd_bus_message *msg = NULL;
+	sd_bus_error error = SD_BUS_ERROR_NULL;
+    sd_bus *sysbus = get_system_bus();
```

Comment:
I'd avoid exposing the system bus; instead, we should rely on the already existent way to do that (async):
```
SYSBUS_ARG_REPLY(...)
```
Basically, the only line that should need a change is:
```
USERBUS_ARG_REPLY(pm_args, parse_bus_reply, &pm_inh_token,
                            "org.freedesktop.PowerManagement.Inhibit",
                            "/org/freedesktop/PowerManagement/Inhibit",
                            "org.freedesktop.PowerManagement.Inhibit",
                            "Inhibit");
```
That should become
```
SYSBUS_ARG_REPLY(pm_args, parse_bus_reply, &pm_inh_token, "org.freedesktop.login1", "/org/freedesktop/login1", "org.freedesktop.login1.Manager", Inhibit);
```
and properly manage the reply message in the `parse_bus_reply` callback :)

Reply:
See https://github.com/FedeDP/Clight/blob/master/src/modules/upower.c#L97 for an example :)